Feature/sc 266/iamport

JunPyoL22 opened the pull request:
### Summary
- 아임포트 결제모듈이 호출되지 않는 이슈를 수정
- 강의시청페이지에 헤더가 나타나는 현상 수정
- 기존 로그인폼 사용하는 부분 제거
- PG사 추가 총8개 지원

### AS-IS
- 아임포트 결제모듈이 호출되지 않는 이슈를 수정
    - 일부 결제관련 정보가 부족하여 아임포트 결제모듈이 호출되지 않는 이슈가 존재
- 강의시청페이지에 헤더가 나타나는 현상 수정
    - 강의시청페이지에 헤더가 나타나 일부 컴포넌트를 가리는 문제
- 기존 로그인폼 사용하는 부분 제거
    - 새롭게 디자인된 로그인폼을 사용하지않고 기존 로그인폼을 사용하던 부분이 존재
- PG사 추가
   - 기존 KG이니시스만 지원
   
### TO-BE
- 아임포트 결제모듈이 호출되지 않는 이슈를 수정
    - 일부 결제관련 정보가 부족하여 아임포트 결제모듈이 호출되도록 변경
- 강의시청페이지에 헤더가 나타나는 현상 수정
    - 강의시청페이지에 헤더가 나타나 일부 컴포넌트를 가리지 않도록 헤더 컴포넌트를 hidden으로 처리
- 기존 로그인폼 사용하는 부분 제거
    - 새롭게 디자인된 로그인폼을 사용하지않고 기존 로그인폼을 사용하던 부분을 제거하고 새로 디자인된 로그인 페이지로 이동시킴
- PG사 추가
   - KG 이니시스외 8개 추가
    
 ### Comment
  - 경엽님이 추가 수정하신 결제페이지 결제카트 Tag 정렬되지 않던 것을 수정하였습니다.
  - 또한 결제페이지 결제카트에 있던 체크박스가 수행하는 기능이 없어서 제거하였습니다.

JunPyoL22 committed:
Fix: 로그인 폼 제거
-디자인된 로그인 적용에 따른 기존 로그인폼을 제거
-로그인 상태가 아닌 경우 alert 창을 보여주고 로그인 페이지로 이동시킴

JunPyoL22 committed:
Design: 헤더와body 사이에 일정한 룸 추가
- 데스크탑 화면에서 헤더와 컨텐츠 바디가 너무 붙어 있는 현상 수정

JunPyoL22 committed:
Fix: 로그인 폼 제거
-디자인된 로그인 적용에 따른 기존 로그인폼을 제거
-로그인 상태가 아닌 상태에서 수강신청 또는 결제하기 버튼 클릭시 alert 창을 보여주고 로그인 페이지로 이동시킴

JunPyoL22 committed:
Fix: 로그인 폼 제거
-디자인된 로그인 적용에 따른 기존 로그인폼을 제거
-로그인 상태가 아닌 상태에서 수강신청 또는 결제하기 버튼 클릭시 alert 창을 보여주고 로그인 페이지로 이동시킴

JunPyoL22 committed:
FIx: 미사용 모듈 제거

JunPyoL22 committed:
FIx: checkbox UI 커맨트 처리
- 체크박스 기능이 없어 결제모듈 심사 진행시 정상 작동하지 않는 사이트라는 인상을 줄 수 있으므로 임시 삭제함

JunPyoL22 committed:
Design: 결제카트 디자인 수정
- TAG 정렬 수정
- hard 코딩된 정보를 Database 데이터로 연동할 수 있도록 수정

JunPyoL22 committed:
Fix: 결제모달이 나타나지 않는 이슈 수정
- 결제방법 선택모달이 나타난 상태에서 결제방식을 클릭하면 결제모달이 보여질 수 있도록 함
- 결제방법 선택모달에서 결제방식을 클릭하면 자동적으로 모달이 사라지도록 변경

JunPyoL22 committed:
Fix: 결제상품명(ProductName)을 MyOrder 인터페이스에 추가

JunPyoL22 committed:
Fix: 결제성공 후 로직과 결제 실패 후 로직을 추가
 - sucessPayment, failPayment 함수를 PaymentMethodModal에Props로 넘겨준다.
 - 결제방법 모달에 '신용카드' 방식만 보이도록 추가적으로 수정

JunPyoL22 committed:
Fix: 결제모듈이 정상적으로 호출될 수 있도록 프런트엔드에서 임의로 상품 금액을 100원으로 설정

JunPyoL22 committed:
Fix: 기존 로그인 폼 제거와 관련한 변경에서 미반영사항 추가

JunPyoL22 committed:
Fix: staging 환경 배포 방식 변경

JunPyoL22 committed:
Fix: 로그인 관련 alert 문구 수정

JunPyoL22 committed:
Fix: 로그인 폼 제거 -디자인된 로그인 적용에 따른 기존 로그인폼을 제거
- 로그인 상태가 아닌 상태에서 수강신청 또는 결제하기 버튼 클릭시 alert 창을 보여주고 로그인 페이지로 이동시킴

JunPyoL22 committed:
Feat: 릴리즈노트

JunPyoL22 committed:
Fix: 강의시청 페이지에서 헤더가 나타나는 문제 수정

JunPyoL22 committed:
Update: package.json 버전 정보

lkylove23 commented:
> 또한 결제페이지 결제카트에 있던 체크박스가 수행하는 기능이 없어서 제거하였습니다.

준표님, 새 결제페이지 디자인시안에도 체크박스가 있습니다. 나중에 필요할때 추가할께요 :)

lkylove23 commented:
준표님, 감사드립니다.

다음에는 PR 제목을 convention(Issue ID: Subject)에 맞춰서 작성하면 어떨까요?

SC-266: 아임포트 가입을 위한 FE 결제모듈 작업

(마음은 정말 이해하지만 ^^) 가능한 하나의 주제를 PR에서 다루면 좋겠습니다. 그래도 commit을 남겨주셔서 나중에 검색할수있겠네요.

lkylove23 commented on src/components/paymentCart/paymentCart.tsx:
```diff
@@ -42,22 +42,16 @@ const CourseCard = ({course}: CourseCardProps) => {
-          <div className = {classes.courseTagItem}>
-            Live 
+        <Group spacing = "xs">
+          <div className = {cx(classes.courseTagBoxGreen, classes.courseTagItem)}>
```
@JunPyoL22 Merge classes 감사합니다. 좋은 추세인것 같네요 :)

Conditional class는 처음 보았습니다.

https://www.felixmokross.dev/blog/merging-css-classes

lkylove23 reviewed (approved):
@JunPyoL22 준표님, 수정 감사드립니다. 👍

lkylove23 reviewed (commented):
@JunPyoL22  p1: 결제방법 모달에 신용카드 외에 어떤것들이 있었지요? 핸드폰 소액결제와 현금결제(?) 도 필요할수도 있습니다.

jplee-swk commented:
@lkylove23 기존 코드는 신용카드 가상계좌 두가지 방법중 선택하도록 되어 있었습니다. 어떤 것을 지원할지는 저희가 정해야 할것 같고 일단 확실한 사항만 포함되도록 수정하였습니다.

JunPyoL22 committed:
Feat: 결제방식지원 추가
- 실시간계좌이체, 가상계좌, 휴대폰소액결제 방식 추가

JunPyoL22 committed:
Fix: 결제 실패 메세시 상세화
- 결제 실패시 실패 사유를 고객에게 정확히 안내하기 위함

JunPyoL22 commented:
@lkylove23 결제방식 3가지를 추가(휴대폰소액결제, 가상계좌, 실시간계좌이체) 하였습니다. 

추가적으로 현재 kg이니시스 결제 테스트 모드로 (슈퍼코딩루트 아이디로 아임포트 로그인후 kg 이니시스 테스트 모드로 설정함) 설정되어 있습니다.

JunPyoL22 committed:
Fix: PR시 staging에 배포될 수 있도록 수정
 - PR 내용을 직접 staging 환경에서 리뷰어가 확인할 수 있도록 하기 위함

JunPyoL22 committed:
Refactor: 결제방식 모달 컴포넌트 인터페이스 수정
- 결제 유저플로우에서 PG사 선택이 추가되면서 변경된 결제플로우를 유연하게 구현하기 위한 구조로 변경
- PG사에 따라 상의한 결제방식을 지원함에 따라, PG사별 결제방식을 다르게 하기 위한 더 좋은 구조로 변경

JunPyoL22 committed:
Feat: PG사 선택 모달 컴포넌트 구현

JunPyoL22 committed:
Feat: 결제관련 설정값을 관리하는 설정클래스 추가

JunPyoL22 committed:
Feat: PG사 선택 모달창에 적용할 스타일 추가

JunPyoL22 committed:
Feat: 결제페이지에 PG사 선택 모달창 적용
- 결제모듈 진입 플로우: 결제하기 버튼 > PG사 선택 > 결제수단 선택 > 아임포트 결제모듈

JunPyoL22 committed:
Fix: 릴리즈노트 PG사 선택 모달창 작업 내용 추가

JunPyoL22 commented:
@lkylove23 PG사를 추가하였습니다. 다만 스마일페이 아임포트 문서에 확실치 않은 것이 있어서 노션에 있던 추가목록중 스마일페이는 추가하지 못했습니다.
추가적으로 결제 심사 진행을 위한 서브도메인(http://jobs.supercoding.net) (`staging 환경`)입니다. 해당 주소로 접속하셔서 수강신청하지 않은 아이디로 결제페이지에서 결제하기 버튼을 클릭해보시면 PG사 추가에 따른 결제플로우를 확인해보실 수 있습니다.
